test_requests.py

```diff
@@ -188,6 +188,26 @@ def test_HTTP_302_ALLOW_REDIRECT_GET(self, httpbin):
         assert r.history[0].status_code == 302
         assert r.history[0].is_redirect
 
+    def test_HTTP_302_TOO_MANY_REDIRECTS(self, httpbin):
+        try:
+            requests.get(httpbin('redirect', '50'))
+        except TooManyRedirects as e:
+            assert e.request is not None
```

Comment:
Can we expand on this assertion a bit? For example, check some properties on the request, and confirm it's ok? Similarly, check the response?